agent/consul/autopilot/autopilot.go

```diff
@@ -0,0 +1,460 @@
+package autopilot
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/go-version"
+	"github.com/hashicorp/raft"
+	"github.com/hashicorp/serf/serf"
+)
+
+// Delegate is the interface for the Autopilot mechanism
+type Delegate interface {
+	AutopilotConfig() *Config
+	FetchStats(context.Context, []serf.Member) map[string]*ServerStats
+	IsServer(serf.Member) (bool, *ServerInfo)
```

Comment:
You could ditch the `bool` and just return `nil`. It might be good to have an `error` return and log that from the autopilot side, too, so we can tell if any of those weird error cases ever get hit.